homeassistant/components/enphase_envoy/quality_scale.yaml

```diff
@@ -0,0 +1,96 @@
+rules:
+  # Bronze
+  action-setup:
+    status: done
+    comment: only actions implemented are platform native ones.
+  appropriate-polling:
+    status: done
+    comment: fixed 1 minute cycle based on Enphase Envoy device characteristics
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
```

Comment:
```py
    with pytest.raises(KeyError) as ex:
        await hass.config_entries.flow.async_init(
            DOMAIN,
            context={"source": SOURCE_ZEROCONF},
            data=zeroconf.ZeroconfServiceInfo(
```
I don't believe this should be able to raise a KeyError Shouldn't we abort the flow?

Reply:
Probably, it's testing the case the Envoy sends garbled property names in discovery.